### ISSUE TITLE
Reduce the amount error event logging

### ISSUE THREAD

**Issue by baron1405** [jshiell/checkstyle-idea#218]
Checkstyle-IDEA issues errors into the IntelliJ event log anytime it has any problem scanning any file. Most of these are for non-java files (e.g. JSON, groovy). I am not sure why Checkstyle-IDEA is scanning non-java files but in any case it would be great if these issues were suppressed. It is rather annoying to have the event log icon flashing all the time with Checkstyle-IDEA scanning issues. Perhaps I have something mis-configured.

Thanks for your help!

IntelliJ Ultimate 15.0.3
Checkstyle-IDEA 4.23.0
RHEL 6.6

**Comment by jshiell** [comment]
Do you have 'Scan non-Java files' toggled on in the plugin preferences? That's the only way it *should* scan non-Java files (although, of course, there's always the chance of bugs).

**Comment by baron1405** [comment]
The "Scan non-Java files" is not checked. I cleared my events and will see if I can characterize more accurately when this occurs.

**Comment by baron1405** [comment]
Just reproduced this by editing a Groovy file under src/main/groovy. This is a Gradle project imported into IntelliJ 15.0.3. Checkstyle is enabled but "Scan non-Java files" is unchecked. The error in the event log is:

10:13:35 AM CheckStylePluginException: CheckStyle threw an exception when scanning: LiIntelliJPlugin.groovy: Couldn't create Checker from LinkedIn Style (ligradle-intellij-plugin_trunk)

**Comment by jshiell** [comment]
Thanks for that. Is there any stack trace? Also, it'd be very helpful if you could attach your rules file. Thank you!

**Comment by baron1405** [comment]
I am still not clear why a Groovy file is being scanned but the stack trace is pointing to a problem that is likely unrelated to the type of file being scanned. The trace is:
```
2016-02-29 14:04:44,729 [1553329933]  ERROR - heckstyle.CheckStyleInspection - CheckStyle threw an exception when scanning: LiIntelliJPlugin.groovy
org.infernus.idea.checkstyle.exception.CheckStylePluginException: Couldn't create Checker from LinkedIn Style (ligradle-intellij-plugin_trunk)
        at org.infernus.idea.checkstyle.checker.CheckerFactory.checker(CheckerFactory.java:70)
        at org.infernus.idea.checkstyle.checker.CheckerFactory.checker(CheckerFactory.java:49)
        at org.infernus.idea.checkstyle.CheckStyleInspection.inspectFile(CheckStyleInspection.java:85)
        at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$checkFile$26(CheckStyleInspection.java:58)
        at org.infernus.idea.checkstyle.CheckStyleInspection$$Lambda$33/270877528.call(Unknown Source)
        at com.intellij.openapi.application.impl.ApplicationImpl$9.call(ApplicationImpl.java:391)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
        at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:55)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Could not load configuration
        at org.infernus.idea.checkstyle.checker.CheckerFactory.createChecker(CheckerFactory.java:149)
        at org.infernus.idea.checkstyle.checker.CheckerFactory.getOrCreateCachedChecker(CheckerFactory.java:84)
        at org.infernus.idea.checkstyle.checker.CheckerFactory.checker(CheckerFactory.java:63)
        ... 10 more
Caused by: java.lang.IllegalArgumentException: No enum constant com.puppycrawl.tools.checkstyle.api.SeverityLevel.PROPERTY 'CHECKSTYLE.SEVERITY' HAS NO VALUE DEFINED IN THE CONFIGURATION.
        at java.lang.Enum.valueOf(Enum.java:238)
        at com.puppycrawl.tools.checkstyle.api.SeverityLevel.getInstance(SeverityLevel.java:67)
        at com.puppycrawl.tools.checkstyle.ConfigurationLoader$InternalLoader.endElement(ConfigurationLoader.java:484)
        at org.apache.xerces.parsers.AbstractSAXParser.endElement(Unknown Source)
        at org.apache.xerces.impl.dtd.XMLNSDTDValidator.endNamespaceScope(Unknown Source)
        at org.apache.xerces.impl.dtd.XMLDTDValidator.handleEndElement(Unknown Source)
        at org.apache.xerces.impl.dtd.XMLDTDValidator.endElement(Unknown Source)
        at org.apache.xerces.impl.XMLNSDocumentScannerImpl.scanEndElement(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
        at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
        at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
        at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
        at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
        at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
        at com.puppycrawl.tools.checkstyle.api.AbstractLoader.parseInputSource(AbstractLoader.java:99)
        at com.puppycrawl.tools.checkstyle.ConfigurationLoader.parseInputSource(ConfigurationLoader.java:153)
        at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:232)
        at org.infernus.idea.checkstyle.checker.CheckerFactoryWorker.run(CheckerFactoryWorker.java:41)
```
The complaint is about the severity property that we have defined in our config file with a default value. It appears that the default value is not being honored. Here is an excerpt of the config file:
```
<module name="Checker">
  <property name="severity" value="warning"/>
  <property name="fileExtensions" value="java"/>

  <module name="TreeWalker">
    ...
  </module>
</module>
```
Notice the "severity" property has a default value of "warning".

The complete config file is:
```
<module name="Checker">
  <property name="severity" value="warning"/>
  <property name="fileExtensions" value="java"/>

  <module name="TreeWalker">
    <module name="SuppressWarningsHolder"/>
    <module name="FileContentsHolder"/>

    <!-- ANNOTATIONS -->

    <!-- No trailing empty parenthesis or commas -->
    <module name="AnnotationUseStyle">
      <property name="elementStyle" value="ignore"/>
    </module>
    <!-- Ensure @Override is present when {@inheritDoc} Javadoc tag is present -->
    <module name="MissingOverride"/>
    <!-- Package level annotations belong in package-info.java -->
    <module name="PackageAnnotation"/>

    <!-- BLOCKS -->

    <!-- Block opening brace on same line -->
    <module name="LeftCurly">
      <property name="option" value="eol"/>
    </module>
    <!-- Block closing brace for else, catch, finally on same line -->
    <module name="RightCurly">
      <property name="option" value="same"/>
    </module>
    <!-- Always use braces even if optional -->
    <module name="NeedBraces"/>

    <!-- CLASS DESIGN -->

    <!-- Classes containing only static methods should not have a public constructor -->
    <module name="HideUtilityClassConstructor"/>

    <!-- CODING -->

    <!-- Use Java style array declarations (e.g. String[] names), not C style (e.g. String names[]) -->
    <module name="ArrayTypeStyle"/>
    <!-- If covariant equals defined, standard equals must also be defined -->
    <module name="CovariantEquals"/>
    <!-- Switch 'default' case must appear last -->
    <module name="DefaultComesLast"/>
    <!-- Override equals and hashCode together -->
    <module name="EqualsHashCode"/>
    <!-- No fall through in switch cases, even the last one -->
    <module name="FallThrough">
      <property name="checkLastCaseGroup" value="true"/>
    </module>
    <!-- Do not perform assignments embedded within expressions -->
    <module name="InnerAssignment"/>
    <!-- Switch statements must have a 'default' case -->
    <module name="MissingSwitchDefault"/>
    <!-- Do not modify the 'for' loop control variable -->
    <module name="ModifiedControlVariable"/>
    <!-- Each variable delcaration must be on a separate line -->
    <module name="MultipleVariableDeclarations"/>
    <!-- Each statement (i.e. code terminated by a semicolon) must be on a separate line -->
    <module name="OneStatementPerLine"/>
    <!-- Classes must have an explicit package declaration -->
    <module name="PackageDeclaration"/>
    <!-- Do not test boolean expressions against the values true or false -->
    <module name="SimplifyBooleanExpression"/>
    <!-- Do not test for boolean conditions and return the values true or false -->
    <module name="SimplifyBooleanReturn"/>
    <!-- Do not use '==' to compare string against a literal; use 'equals' -->
    <module name="StringLiteralEquality"/>
    <!-- Use 'L' with long literals -->
    <module name="UpperEll"/>

    <!-- IMPORTS -->

    <!-- No imports statements using '*' notation except static imports -->
    <module name="AvoidStarImport">
      <property name="allowStaticMemberImports" value="true"/>
    </module>
    <!-- Do not import 'sun' packages -->
    <module name="IllegalImport"/>
    <!-- Do not duplicate import statements -->
    <module name="RedundantImport"/>
    <!-- Eliminate unused imports -->
    <module name="UnusedImports"/>

    <!-- JAVADOC COMMENTS -->

    <!-- If you have a Javadoc comment, make sure it is properly formed -->
    <module name="JavadocStyle">
      <property name="checkFirstSentence" value="false"/>
    </module>

    <!-- NAMING CONVENTIONS -->

    <!-- Generic parameters for a class must be uppercase letters (e.g. <T>, <OLD>) -->
    <module name="ClassTypeParameterName">
      <property name="format" value="^[A-Z]+$"/>
    </module>
    <!-- Constants must be all uppercase letters separated by underscores -->
    <module name="ConstantName">
      <property name="format" value="^(_?log)|([A-Z][A-Z0-9]*(_[A-Z0-9]+)*)$"/>
    </module>
    <!-- Local variables must be camel case starting with lowercase letter -->
    <module name="LocalFinalVariableName"/>
    <module name="LocalVariableName"/>
    <!-- Member variables must be camel case starting with an underscore and lowercase letter -->
    <module name="MemberName">
      <property name="format" value="^_[a-z][a-zA-Z0-9]*$"/>
    </module>
    <!-- Method name must be camel case starting with a lowercase letter -->
    <module name="MethodName"/>
    <!-- Generic parameters for a method must be uppercase letters separated by underscore (e.g. <V>, <NEW>) -->
    <module name="MethodTypeParameterName">
      <property name="format" value="^[A-Z]+(_[A-Z]+)*$"/>
    </module>
    <!-- Package name must be all lowercase letters separated by periods -->
    <module name="PackageName">
      <property name="format" value="^[a-z]+(\.[a-z][a-z0-9]*)*$"/>
    </module>
    <!-- Parameters must be camel case starting with a lowercase letter -->
    <module name="ParameterName"/>
    <!-- Static variables must be camel case starting with an underscore and lowercase letter -->
    <module name="StaticVariableName">
      <property name="format" value="^_[a-z][a-zA-Z0-9]*$"/>
    </module>
    <!-- Type names must be camel case starting with an uppercase letter -->
    <module name="TypeName"/>

    <!-- LENGTHS -->

    <!-- Desired line length is 120 but allow some overrun beyond that -->
    <module name="LineLength">
      <property name="max" value="160"/>
      <message key="maxLineLen" value="Line is longer than {0,number,integer} characters (found {1,number,integer}). Try to keep lines under 120 characters."/>
    </module>

    <!-- WHITESPACE -->

    <module name="GenericWhitespace"/>
    <module name="MethodParamPad"/>
    <module name="NoWhitespaceAfter">
      <property name="tokens" value="BNOT,DEC,DOT,INC,LNOT,UNARY_MINUS,UNARY_PLUS"/>
    </module>
    <module name="NoWhitespaceBefore"/>
    <module name="OperatorWrap"/>
    <module name="ParenPad"/>
    <module name="TypecastParenPad">
      <property name="tokens" value="RPAREN,TYPECAST"/>
    </module>
    <module name="WhitespaceAfter"/>
    <module name="WhitespaceAround"/>

    <!-- Do not allow meaningless, IDE generated parameter names -->
    <module name="RegexpSinglelineJava">
      <property name="format" value="[\s]+arg[\d]+[,\)]"/>
      <property name="message" value="Replace argN with a meaningful parameter name"/>
    </module>
  </module>

  <!-- Do not allow tab characters in source files -->
  <module name="FileTabCharacter"/>

  <!-- Ensure parameter and exception names are present on @param and @throws tags -->
  <module name="RegexpSingleline">
    <property name="format" value="\*[\s]*@(throws|param)[\s]*$"/>
    <property name="message" value="Missing parameter or exception name"/>
  </module>
  <!-- IDE generated code must be reviewed by developer -->
  <module name="RegexpSingleline">
    <property name="format" value="\/\/[\s]*TODO[\s]+Auto-generated"/>
    <property name="message" value="Replace IDE generated code with real implementation"/>
  </module>
  <!-- Detect commonly misspelled Javadoc tags -->
  <module name="RegexpSingleline">
    <property name="format" value="\*[\s]*@(params|throw|returns)[\s]+"/>
    <property name="message" value="Correct misspelled Javadoc tag"/>
  </module>

  <!-- Read checker suppressions from a file -->
  <module name="SuppressionFilter">
    <property name="file" value="${config_loc}/suppressions.xml"/>
  </module>
  <!-- Allow Checkstyle warnings to be suppressed using trailing comments -->
  <module name="SuppressWithNearbyCommentFilter"/>
  <!-- Allow Checkstyle warnings to be suppressed using block comments -->
  <module name="SuppressionCommentFilter"/>
  <!-- Allow SuppressWarnings annotation to suppress Checkstyle issues -->
  <module name="SuppressWarningsFilter"/>
</module>
```

**Comment by jshiell** [comment]
Sorry about this, March got away on me.

I believe you're correct that it's not related to it being a Groovy file - it looks like it's related to parsing the configuration. The interesting bit is that I can't reproduce this - at least on the latest version of the plugin/Checkstyle. I've also tried running it directly against Checkstyle 6.14.1, which should match the version you were using, but it still seems to work happily.

In theory the behaviour of the enum parse could change depending on your locale, but this seems an extreme edge case. This could probably be ruled out by setting the value to WARNING rather than warning.

I did have to make one small change and add the DTD to the top of the XML file mind, or Checkstyle just refused to even parse it.
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
``` 

I presume you see the same behaviour on the latest version of the plugin/Checkstyle?
